src/ArchiveAdmin.php

```diff
@@ -130,48 +132,24 @@ public function getEditForm($id = null, $fields = null)
      */
     public static function createArchiveGridField($title, $class)
     {
+        if (!is_a($class, DataObject::class, true)) {
+            throw new InvalidArgumentException("Class $class in not a DataObject");
+        }
         $config = GridFieldConfig_Base::create();
         $config->removeComponentsByType(VersionedGridFieldState::class);
         $config->removeComponentsByType(GridFieldFilterHeader::class);
         $config->addComponent(new GridFieldDetailForm);
         $config->addComponent(new GridFieldViewButton);
         $config->addComponent(new GridFieldRestoreAction);
         $config->addComponent(new GridField_ActionMenu);
-
-        $singleton = singleton($class);
```

Comment:
This blocks been moved to a private method so that it can be unit tested in isolation